### PR TITLE
Deprecate delete_retention_ms and replace with log_retention_ms

### PR DESCRIPTION
**Pull request by Feediver1**
View the new entry under Deprecated features: https://deploy-preview-161--redpanda-docs-preview.netlify.app/23.3/upgrade/deprecated/

View the deprecation notice in the context of `delete_retention_ms`: https://deploy-preview-161--redpanda-docs-preview.netlify.app/23.3/reference/cluster-properties/#delete_retention_ms

View the new entry for `log_retention_ms`: https://deploy-preview-161--redpanda-docs-preview.netlify.app/23.3/reference/cluster-properties/#log_retention_ms

Updates in `retention.ms`: https://deploy-preview-161--redpanda-docs-preview.netlify.app/23.3/reference/topic-properties/#retentionms

Update in topic<->cluster corresponding properties table: https://deploy-preview-161--redpanda-docs-preview.netlify.app/23.3/reference/topic-properties/

### CODE REVIEW

File: modules/reference/pages/cluster-properties.adoc [hunk 1]
```diff
@@ -1234,6 +1220,16 @@ The topic property xref:./topic-properties.adoc#messagetimestamptype[`message.ti
...
+=== log_retention_ms
+
+The amount of time to keep a log file before deleting it (in milliseconds). If not set, the value in log.retention.minutes is used. If set to -1, no time limit is applied. This is a cluster-wide default when a topic does not set or disable xref:./topic-properties.adoc#retentionms[`retention.ms`].
+
+*Units*: milliseconds
+
+*Default*: null
+
+---
+
```
Comment (andijcr) [inline modules/reference/pages/cluster-properties.adoc]: https://github.com/redpanda-data/redpanda/blob/455c2f19bbfc27c76cec4e99eb0cfc49260b2fa4/src/v/config/configuration.cc#L708-L715

need to remove the reference to log.retention.minutes (we don't have this property) and add the Default of 1 week (604800000 ms)

```suggestion
=== log_retention_ms

The amount of time to keep a log file before deleting it (in milliseconds). If set to -1, no time limit is applied. This is a cluster-wide default when a topic does not set or disable xref:./topic-properties.adoc#retentionms[`retention.ms`].

*Units*: milliseconds

*Default*: 1 week

---

```

File: modules/reference/pages/topic-properties.adoc [hunk 2]
```diff
@@ -15,7 +15,7 @@ Many topic-level properties have corresponding xref:manage:cluster-maintenance/c
 | xref:./cluster-properties.adoc#retention_bytes[`retention_bytes`]
```
Comment (andijcr) [inline modules/reference/pages/topic-properties.adoc:15]: unrelated, but this makes me realize that for consistency we could also rename the retention_bytes cluster property to log_retention_bytes.
but the change would be more than cosmetic so not something that we can do right now.

**Comment by Feediver1** [comment]
@Deflaimun Can you help me understand why this PR pulling in all of these extraneous changes/other PRs?

File: modules/reference/pages/topic-properties.adoc (hunk 2 above)
Comment (Feediver1) [reply modules/reference/pages/topic-properties.adoc:15]: Hmmm...sounds like it cannot be changed yet? Or at least not in this PR. So leaving as is for now.

File: modules/reference/pages/topic-properties.adoc [hunk 3]
```diff
@@ -15,7 +15,7 @@ Many topic-level properties have corresponding xref:manage:cluster-maintenance/c
...
 
 | <<retentionms,`retention.ms`>>
-| xref:./cluster-properties.adoc#delete_retention_ms[`delete_retention_ms`]
+| xref:./cluster-properties.adoc#_log_retention_ms[`log_retention_ms`]
```
Comment (Feediver1) [inline modules/reference/pages/topic-properties.adoc]:
```suggestion
| xref:./cluster-properties.adoc#log_retention_ms[`log_retention_ms`]
```

File: modules/reference/pages/topic-properties.adoc [hunk 4]
```diff
@@ -192,7 +192,7 @@ When both size-based (`retention.bytes`) and time-based (`retention.ms`) retenti
 
 A time-based retention limit that configures the maximum duration that a log's segment file for a topic is retained before it becomes eligible to be cleaned up. To consume all data, a consumer of the topic must read from a segment before its `retention.ms` elapses, otherwise the segment may be compacted and/or deleted. If a non-positive value, no per-topic limit is applied.
 
-If `retention.ms` is set to a positive value, it overrides the cluster property xref:./cluster-properties.adoc#delete_retention_ms[`delete_retention_ms`] for the topic.
+If `retention.ms` is set to a positive value, it overrides the cluster property xref:./cluster-properties.adoc#_log_retention_ms[`log_retention_ms`] for the topic.
```
Comment (Feediver1) [inline modules/reference/pages/topic-properties.adoc]:
```suggestion
If `retention.ms` is set to a positive value, it overrides the cluster property xref:./cluster-properties.adoc#log_retention_ms[`log_retention_ms`] for the topic.
```

File: modules/upgrade/pages/deprecated/index.adoc [hunk 5]
```diff
@@ -1,11 +1,15 @@
...
 |===
-| Feature | Deprecated in
+| Feature | Deprecated in | Details
 
 | xref:./cluster-resource.adoc[Cluster and Console custom resources]
 | 23.2.1
+|
+| xref:reference:cluster-properties.adoc#_delete_retention_ms[`delete_retention_ms`]
+| 23.3
+| Use xref:reference:cluster-properties.adoc#_log_retention_ms[`log_retention_ms`] instead.
```
Comment (Feediver1) [inline modules/upgrade/pages/deprecated/index.adoc]:
```suggestion
| Use xref:reference:cluster-properties.adoc#log_retention_ms[`log_retention_ms`] instead.
```

File: modules/upgrade/pages/deprecated/index.adoc [hunk 6]
```diff
@@ -1,11 +1,15 @@
...
+This index helps you to identify deprecated features in Redpanda releases and plan necessary upgrades or modifications.
 
 |===
-| Feature | Deprecated in
+| Feature | Deprecated in | Details
 
 | xref:./cluster-resource.adoc[Cluster and Console custom resources]
 | 23.2.1
+|
+| xref:reference:cluster-properties.adoc#_delete_retention_ms[`delete_retention_ms`]
```
Comment (Feediver1) [inline modules/upgrade/pages/deprecated/index.adoc]:
```suggestion
| xref:reference:cluster-properties.adoc#delete_retention_ms[`delete_retention_ms`]
```

**Review by andijcr — APPROVED** [review]

**Review by greg-szrama — APPROVED** [review]
Looks good!

File: modules/reference/pages/cluster-properties.adoc [hunk 7]
```diff
@@ -1148,23 +1148,9 @@ NOTE: `cloud_storage_cache_size_percent` overrides `cloud_storage_cache_size`.
...
-* `delete.retention.ms` sets the time-based retention limit for tombstone markers of a compacted topic.
-====
-
-*Units*: milliseconds
-
-*Default*: 604800000 (1 week)
+{badge-deprecated}
 
-*Restart required*: no
+IMPORTANT: Starting in version 23.3, `delete_retention_ms` is deprecated. Use xref:upgrade:deprecated/cluster-resource.adoc[log_retention_ms] instead.
```
Comment (Feediver1) [inline modules/reference/pages/cluster-properties.adoc]:
```suggestion
IMPORTANT: Starting in version 23.3.1, `delete_retention_ms` is deprecated. Use xref:upgrade:deprecated/cluster-resource.adoc[log_retention_ms] instead.
```

File: modules/upgrade/pages/deprecated/index.adoc [hunk 8]
```diff
@@ -1,11 +1,15 @@
...
 
 |===
-| Feature | Deprecated in
+| Feature | Deprecated in | Details
 
 | xref:./cluster-resource.adoc[Cluster and Console custom resources]
 | 23.2.1
+|
+| xref:reference:cluster-properties.adoc#delete_retention_ms[`delete_retention_ms`]
+| 23.3
```
Comment (Feediver1) [inline modules/upgrade/pages/deprecated/index.adoc]:
```suggestion
| 23.3.1
```

**Comment by Feediver1** [comment]
Has a doc team member approval, but @greg-szrama needs to be added to the documentation group in GH. Merging.

**Pull request merged**